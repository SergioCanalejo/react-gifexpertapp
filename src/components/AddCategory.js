import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const handleInput = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const newCategory = inputValue.trim();
    if (newCategory.length) {
      setCategories((categories) => [...new Set([newCategory, ...categories])]);
    }
    setInputValue("");
  };

  const [inputValue, setInputValue] = useState("");
  return (
    <form onSubmit={handleSubmit}>
      <h2>Add Category</h2>
      <input type="text" value={inputValue} onChange={handleInput} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
