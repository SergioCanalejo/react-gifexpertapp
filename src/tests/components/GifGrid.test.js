import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Testing GifGrid component', () => {
  useFetchGifs.mockReturnValue({
    gifs: [],
    loading: true,
  });
  const category = 'One Piece';
  const mockData = [
    {
      id: 'ABC',
      url: 'https://localhost/dummy.jpg',
      title: 'Dummy img',
    },
  ];

  test('Must show the component correctly', () => {
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Must show items from useFetchGifs', () => {
    useFetchGifs.mockReturnValue({ gifs: mockData, loading: false });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('GifGridItem').length).toBe(mockData.length);
  });
});
