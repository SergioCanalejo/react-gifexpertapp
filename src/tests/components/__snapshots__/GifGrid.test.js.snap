// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component Must show the component correctly 1`] = `
<Fragment>
  <h3>
    One Piece
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
